backend/src/main/java/net/vidux/camhub/discovery/StreamGobbler.java

```diff
@@ -0,0 +1,23 @@
+package net.vidux.camhub.discovery;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.function.Consumer;
+
+class StreamGobbler implements Runnable {
+  private InputStream inputStream;
+  private Consumer<String> consumer;
+
+  public StreamGobbler(InputStream inputStream, Consumer<String> consumer) {
+    this.inputStream = inputStream;
+    this.consumer = consumer;
+  }
+
+  @Override
+  public void run() {
+    if (inputStream != null && consumer != null) {
+      new BufferedReader(new InputStreamReader(inputStream)).lines().forEach(consumer);
```

Comment:
Sonar rule  java:S1943. Classes and methods that rely on the default system encoding should not be used.